### PR TITLE
SOV-1120: Redemptions history frame

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-1120

### CODE REVIEW

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/hooks/useGetRedemptionsHistory.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  GetRedemptionsDocument,
+  Redemption_Filter,
+} from './../../../../utils/graphql/zero/generated';
+
+const zeroClient = new ApolloClient({
+  uri: graphZeroUrl,
+  cache: new InMemoryCache({
+    resultCaching: false,
+  }),
+});
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/hooks/useGetRedemptionsHistory.ts]: we can move this to another file and import it from there

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/hooks/useGetRedemptionsHistory.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        redeemer: account,
+      } as Redemption_Filter,
+    }),
+    [page, orderOptions, pageSize, account],
+  );
+
+  const { loading, data } = useQuery(GetRedemptionsDocument, {
+    variables: redemptionConfig,
+    client: zeroClient,
+  });
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/hooks/useGetRedemptionsHistory.ts]: any reason we stopped using generated hooks?

**Review by Rickk137 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        id: 'sequenceNumber',
+        title: t(translations.redemptionsHistory.table.timestamp),
+        cellRenderer: (item: Redemption) =>
+          dateFormat(item.transaction.timestamp),
+        sortable: true,
+      },
+      {
+        id: 'zusdRedeemed',
+        title: t(translations.redemptionsHistory.table.zusdRedeemed),
+        cellRenderer: (item: Redemption) => renderZUSDRedeemed(item),
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: cellRenderer: renderZUSDRedeemed

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+      },
+      {
+        id: 'rbtcReceived',
+        title: t(translations.redemptionsHistory.table.rbtcReceived),
+        cellRenderer: (item: Redemption) => renderRBTCReceived(item),
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: cellRenderer:renderRBTCReceived

**Review by Rickk137 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+        onExportEnd={() => setPageSize(DEFAULT_PAGE_SIZE)}
+        disabled={!redemptions || redemptions.length === 0}
+      />
+      <div className="bg-gray-80 py-4 px-4 rounded">
+        <Table
+          setOrderOptions={setOrderOptions}
+          orderOptions={orderOptions}
+          columns={columns}
+          rows={redemptions}
+          rowTitle={row => generateRowTitle(row)}
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: rowTitle={generateRowTitle}

**Review by Rickk137 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+          isLoading={loading}
+          className="bg-gray-80 text-gray-10 lg:px-6 lg:py-4"
+          noData={t(translations.common.tables.noData)}
+          dataAttribute="transaction-history-table"
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: please update the dataAttribute and use applyDataAttr function

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        />
+        <Pagination
+          page={page}
+          className="lg:pb-6 mt-3 lg:mt-6 justify-center lg:justify-start"
+          onChange={onPageChange}
+          itemsPerPage={pageSize}
+          isNextButtonDisabled={isNextButtonDisabled}
+          dataAttribute="transaction-history-pagination"
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: please update the dataAttribute and use applyDataAttr function

**Review by Rickk137 — CHANGES_REQUESTED** [review]

**Review by Rickk137 — APPROVED** [review]

**Review by creed-victor — APPROVED** [review]

File: apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+    setPageSize(EXPORT_RECORD_LIMIT);
+
+    return redemptions.map(tx => ({
+      timestamp: dateFormat(tx.transaction.timestamp),
+      zusdRedeemed: tx.tokensActuallyRedeemed,
+      rbtcReceived: tx.collateralRedeemed,
+      redemptionFee: tx.fee,
+      transactionID: tx.transaction.id,
+    }));
+  }, [t, redemptions, addNotification]);
```
Comment (Rickk137) [inline apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: This is not quite right. If you think about it you are setting the setPageSize to 500 and then creating the CSV. The data takes time to be fetched so you end up getting the current pageSize (20) items exported. This needs to be change like the way i did for Collateral surplus withdrawals.
https://github.com/DistributedCollective/sovryn-dapp/pull/170/files#diff-5bf47db7ed7ad9aabfbaaca83181257902b6aa24880641c26d528a7edb578925R122-R152

Comment (Rickk137) [reply apps/frontend/src/app/3_organisms/RedemptionsHistoryFrame/RedemptionsHistoryFrame.tsx]: the same thing apply for TransactionHistoryFrame

**Pull request merged**